src/Common/src/System/Net/Mail/MailBnfHelper.cs

```diff
@@ -0,0 +1,419 @@
+                    throw new FormatException(SR.Format(SR.MailHeaderFieldInvalidCharacter, data[offset]));
+                }
+            }
+            if (doesntRequireQuotes)
+            {
+                localBuilder.Append(data, start, offset - start);
+                return (builder != null ? null : localBuilder.ToString());
+            }
+            throw new FormatException(SR.MailHeaderFieldMalformedHeader);
+        }
+
+        internal static string ReadParameterAttribute(string data, ref int offset, StringBuilder builder)
+        {
+            if (!SkipCFWS(data, ref offset))
+                return null; // 
+
+            return ReadToken(data, ref offset, null);
+        }
+
+        internal static string ReadToken(string data, ref int offset, StringBuilder builder)
+        {
+            int start = offset;
+            for (; offset < data.Length; offset++)
+            {
+                if (data[offset] > Ascii7bitMaxValue)
+                {
+                    throw new FormatException(SR.Format(SR.MailHeaderFieldInvalidCharacter, data[offset]));
+                }
+                else if (!Ttext[data[offset]])
+                {
+                    break;
+                }
+            }
+
+            if (start == offset)
+            {
+                throw new FormatException(SR.Format(SR.MailHeaderFieldInvalidCharacter, data[offset]));
+            }
+
+            return data.Substring(start, offset - start);
+        }
+
+        private static string[] s_months = new string[] { null, "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" };
+
+        internal static string GetDateTimeString(DateTime value, StringBuilder builder)
+        {
+            StringBuilder localBuilder = (builder != null ? builder : new StringBuilder());
+            localBuilder.Append(value.Day);
+            localBuilder.Append(' ');
+            localBuilder.Append(s_months[value.Month]);
+            localBuilder.Append(' ');
+            localBuilder.Append(value.Year);
+            localBuilder.Append(' ');
+            if (value.Hour <= 9)
+            {
+                localBuilder.Append('0');
+            }
+            localBuilder.Append(value.Hour);
+            localBuilder.Append(':');
+            if (value.Minute <= 9)
+            {
+                localBuilder.Append('0');
+            }
+            localBuilder.Append(value.Minute);
+            localBuilder.Append(':');
+            if (value.Second <= 9)
+            {
+                localBuilder.Append('0');
+            }
+            localBuilder.Append(value.Second);
+
+            string offset = TimeZoneInfo.Local.GetUtcOffset(value).ToString();
+            if (offset[0] != '-')
+            {
+                localBuilder.Append(" +");
+            }
+            else
+            {
+                localBuilder.Append(' ');
+            }
+
+            string[] offsetFields = offset.Split(':');
```

Comment:
Rather than a `Split` this could be an `IndexOf` and use the partial `Appends` on the `StringBuilder`. Save a couple of arrays and 2 strings.